### PR TITLE
Add rgb  matrix support for q2 of keychron. 

### PR DESCRIPTION
**Pull request by lalalademaxiya1**
## Description
- Add rgb matrix support and based on ckled2001.
- Delete matrix.c, Because we have replaced the resistor R7 from 10KΩ to 100KΩ on hardware so that we can reuse the boot pin of stm32l432 as a gerneral IO.
- We used mcu stm32l433 to replace stm32l432 temporarily , Because it has the similar configuration with stm32l432 and so that we can make the rule.mk more brief.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/keychron/q2/chconf.h [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 /*
  * This file was auto-generated by:
- *    `qmk chibios-confmigrate -i keyboards/acheron/arctic/chconf.h -r platforms/chibios/common/configs/chconf.h`
+ *    `qmk chibios-confmigrate -i keyboards/keychron/q2/chconf.h -r platforms/chibios/boards/common/configs/chconf.h`
```
Comment (tzarc) [inline keyboards/keychron/q2/chconf.h:19]: Just delete this whole comment block. It's injected when it's been migrated, but serves no purpose when it's a new board. No need to keep it.

File: keyboards/keychron/q2/halconf.h [hunk 2]
```diff
@@ -16,11 +16,11 @@
 
 /*
  * This file was auto-generated by:
- *    `qmk chibios-confmigrate -i keyboards/nk65/halconf.h -r platforms/chibios/QMK_PROTON_C/configs/halconf.h`
+ *    `qmk chibios-confmigrate -i keyboards/keychron/q2/halconf.h -r platforms/chibios/boards/QMK_PROTON_C/configs/halconf.h`
```
Comment (tzarc) [inline keyboards/keychron/q2/halconf.h:19]: Just delete this whole comment block. It's injected when it's been migrated, but serves no purpose when it's a new board. No need to keep it.

File: keyboards/keychron/q2/mcuconf.h [hunk 3]
```diff
@@ -16,12 +16,12 @@
 
 /*
  * This file was auto-generated by:
- *    `qmk chibios-confmigrate -i keyboards/acheron/arctic/mcuconf.h -r platforms/chibios/GENERIC_STM32_F072XB/configs/mcuconf.h`
+ *    `qmk chibios-confmigrate -i keyboards/keychron/q2/mcuconf.h -r platforms/chibios/boards/QMK_PROTON_C/configs/mcuconf.h`
```
Comment (tzarc) [inline keyboards/keychron/q2/mcuconf.h:19]: Just delete this whole comment block. It's injected when it's been migrated, but serves no purpose when it's a new board. No need to keep it.

**Review by tzarc — CHANGES_REQUESTED** [review]
Just some minor cleanup.

File: keyboards/keychron/q2/chconf.h (hunk 1 above)
Comment (lalalademaxiya1) [reply keyboards/keychron/q2/chconf.h:19]: Thanks a lot. I got it.

**Review by drashna — APPROVED** [review]

**Review by tzarc — APPROVED** [review]

**Pull request merged**